x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/SenderService.java

```diff
@@ -71,11 +71,16 @@ public void infer(
         Map<String, Object> taskSettings,
         InputType inputType,
         @Nullable TimeValue timeout,
-        ActionListener<InferenceServiceResults> listener
+        ActionListener<InferenceServiceResults> listener,
+        @Nullable List<String> imageUrls
     ) {
         timeout = ServiceUtils.resolveInferenceTimeout(timeout, inputType, clusterService);
         init();
-        var inferenceInput = createInput(this, model, input, inputType, query, returnDocuments, topN, stream);
+        // TODO: combine images and text inputs into one list?
+        if (input == null) {
+            input = List.of();
+        }
```

Comment:
This is a hacky workaround for the case where a multimodal embedding is performed with only image URLs and no text inputs. The current implementation assumes that there will always be something in the `input` field, but this implementation allows for empty `input` providing that `image_url` is not also empty. Changing to having a single list of inputs which can be either text or image URLs will resolve this problem.